.github/workflows/docs.yml

```diff
@@ -20,73 +28,108 @@ jobs:
-
       - name: Build C module
-        run: make cmodule
+        if: env.MAKE_TARGET
+        run: venv/bin/activate && make $MAKE_TARGET
 
       - name: Build Docs
-        run: make -C docs
+        run: venv/bin/activate && make -C docs
+
+      - name: Wait for others to finish
+        uses: softprops/turnstyle@v1
+        same-branch-only: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.ADMINBOT_TOKEN }}
 
       - name: Checkout docs site
-        if: github.repository_owner == 'tskit-dev'
+        if: github.repository_owner == env.OWNER
         uses: actions/checkout@v2
         with:
-          repository: tskit-dev/msprime-docs
-          token: ${{ secrets.ADMINBOT_DOCS_TOKEN }}
-          path: msprime-docs
+          repository: ${{ env.OWNER }}/${{ env.REPO }}-docs
+          token: ${{ secrets.ADMINBOT_TOKEN }}
+          path: ${{ env.REPO }}-docs
 
-      - name: Check for diff
-        if: github.repository_owner == 'tskit-dev'
+      - name: Check for diff compared to main branch
+        if: github.repository_owner == env.OWNER && github.event.pull_request
         id: diff
         #The html contains commit hashes which we don't want to count as a change
         run: |
-          diff -x .buildinfo -r msprime-docs/main docs/_build/html | grep "^[<>]" | grep -v "<title>\|VERSION:" | grep . && echo "::set-output name=change_detected::true" || echo "NO CHANGES"
+          diff -x .buildinfo -r ${{ env.REPO }}-docs/main docs/_build/html | grep "^[<>]" | grep -v "<title>\|VERSION:" | grep -Ev "\.(css|js)\?" | grep -v "Search\.setIndex" | grep . && echo "::set-output name=change_detected::true" || echo "NO CHANGES"
 
       - name: Copy our docs to the PR specific location
-        if: github.repository_owner == 'tskit-dev' && github.event.pull_request && steps.diff.outputs.change_detected
+        if: github.repository_owner == ${{ env.OWNER }} && steps.diff.outputs.change_detected
+        working-directory: ${{ env.REPO }}-docs
         run: |
-          cd msprime-docs
-          rm -rf ${{github.event.pull_request.number}}
-          cp -r ../docs/_build/html ${{github.event.pull_request.number}}
+          rm -rf ${{ github.event.pull_request.number }}
+          cp -r ../docs/_build/html PR-${{ github.event.pull_request.number }}
 
       - name: Copy our docs to the tag specific location
-        if: github.repository_owner == 'tskit-dev' && !github.event.pull_request
+        if: github.repository_owner == env.OWNER && !github.event.pull_request
+        working-directory: ${{ env.REPO }}-docs
         run: |
-          cd msprime-docs
           export DEST=`echo ${GITHUB_REF} | sed -e "s/refs\/heads\///g" |  sed -e "s/refs\/tags\///g"`
           rm -rf $DEST
           cp -r ../docs/_build/html $DEST
 
-      - name: Commit and push the docs
-        if: github.repository_owner == 'tskit-dev' && (steps.diff.outputs.change_detected || (!github.event.pull_request))
+      - name: Delete closed PRs
```

Comment:
This bit is very clever. But I just want to get some clarification of the behaviour here... If we merge PR `#aaa`, this action will run with the `main` branch trigger (or whatever it's called), deleting the docs for `#aaa`. But if we close PR `#aaa` without merging, this won't run? If someone opens PR `#bbb` after `#aaa` is closed-without-merging, then the `#aaa` docs are deleted, right? Is there also a race condition in here where two instances of the admin bot push commits with the same docs folder deleted?

Reply:
You've got that right - I thought it was probably OK for a closed PR's docs to hang around until the next run of the action.

As for race conditions - if two actions are simultaneously running then who ever got the push in last would wipe out the first actions commit. To mitigate this I've added the `turnstyle` step which waits for other doc actions to finish.

Reply:
Got it! Thanks!

Reply:
:+1: Thanks for looking over these, this stuff is hard to test so a second pair of eyes is great.